### PR TITLE
Feat: get private chat messages

### PR DESCRIPTION
**Pull request by Julieta11**
Closes https://github.com/decentraland/kernel/issues/345

# What? 
Add lazy loading of private conversation

### CODE REVIEW

File: packages/shared/friends/sagas.ts [hunk 1]
```diff
@@ -435,6 +437,34 @@ export function getFriendRequests(request: GetFriendRequestsPayload) {
...
+  const client: SocialAPI | null = getSocialClient(store.getState())
+  if (!client) return
+
+  const socialData: SocialData | undefined = findPrivateMessagingFriendsByUserId(store.getState(), userId)
+  if (!socialData?.conversationId) return
+
+  const ownId = client.getUserId()
+
+  // mark as seen all the messages in the conversation
+  client.markMessagesAsSeen(socialData.conversationId)
```
Comment (jmoguilevsky) [inline packages/shared/friends/sagas.ts:450]: this is a promise, should it be awaited?

File: packages/unity-interface/UnityInterface.ts [hunk 2]
```diff
@@ -349,6 +355,14 @@ export class UnityInterface implements IUnityInterface {
     this.SendMessageToUnity('Main', 'AddFriendRequests', JSON.stringify(addFriendRequestsPayload))
   }
 
+  public UpdateTotalUnseenMessages(updateTotalUnseenMessagesPayload: UpdateTotalUnseenMessagesPayload): void {
+    this.SendMessageToUnity('Main', 'AddFriendRequests', JSON.stringify(updateTotalUnseenMessagesPayload))
```
Comment (jmoguilevsky) [inline packages/unity-interface/UnityInterface.ts:359]:
```suggestion
    this.SendMessageToUnity('Main', 'UpdateTotalUnseenMessages', JSON.stringify(updateTotalUnseenMessagesPayload))
```

File: packages/unity-interface/UnityInterface.ts [hunk 3]
```diff
@@ -349,6 +355,14 @@ export class UnityInterface implements IUnityInterface {
...
+  }
+
+  public UpdateUserUnseenMessages(updateUserUnseenMessagesPayload: UpdateUserUnseenMessagesPayload): void {
+    this.SendMessageToUnity('Main', 'AddFriendRequests', JSON.stringify(updateUserUnseenMessagesPayload))
```
Comment (jmoguilevsky) [inline packages/unity-interface/UnityInterface.ts:363]:
```suggestion
    this.SendMessageToUnity('Main', 'UpdateUserUnseenMessages', JSON.stringify(updateUserUnseenMessagesPayload))
```

File: packages/shared/friends/sagas.ts [hunk 4]
```diff
@@ -435,6 +437,68 @@ export function getFriendRequests(request: GetFriendRequestsPayload) {
...
+  const client: SocialAPI | null = getSocialClient(store.getState())
+  if (!client) return
+
+  const socialData: SocialData | undefined = findPrivateMessagingFriendsByUserId(store.getState(), userId)
+  if (!socialData?.conversationId) return
+
+  const ownId = client.getUserId()
+
+  // get cursor of the conversation located on the given message or at the end of the conversation if there is no given message.
+  const cursorLastMessage = await client.getCursorOnMessage(socialData.conversationId, from, {
```
Comment (jmoguilevsky) [inline packages/shared/friends/sagas.ts:478]: Is this a promise?

Comment (Julieta11) [reply packages/shared/friends/sagas.ts:478]: Yes

File: packages/shared/friends/sagas.ts [hunk 5]
```diff
@@ -435,6 +438,67 @@ export function getFriendRequests(request: GetFriendRequestsPayload) {
...
+  const socialData: SocialData | undefined = findPrivateMessagingFriendsByUserId(store.getState(), userId)
+  if (!socialData?.conversationId) return
+
+  const ownId = client.getUserId()
+
+  // mark as seen all the messages in the conversation
+  await client.markMessagesAsSeen(socialData.conversationId)
+
+  // get total and user's chat unread messages
+  const unreadMessages = client.getConversationUnreadMessages(socialData.conversationId).length
```
Comment (moliva) [inline packages/shared/friends/sagas.ts:454]: How does this work as it seems the previous step was to set every message as seen, do we still get something here?

File: packages/shared/friends/sagas.ts [hunk 6]
```diff
@@ -435,6 +438,67 @@ export function getFriendRequests(request: GetFriendRequestsPayload) {
...
+  const socialData: SocialData | undefined = findPrivateMessagingFriendsByUserId(store.getState(), userId)
+  if (!socialData?.conversationId) return
+
+  const ownId = client.getUserId()
+
+  // get cursor of the conversation located on the given message or at the end of the conversation if there is no given message.
+  const messageId: string | undefined = fromMessageId === null ? undefined : fromMessageId
+  const cursorLastMessage = await client.getCursorOnMessage(socialData.conversationId, messageId, {
+    initialSize: limit,
+    limit: limit
```
Comment (moliva) [inline packages/shared/friends/sagas.ts:482]:
```suggestion
    limit
```

File: packages/shared/friends/sagas.ts (hunk 5 above)
Comment (Julieta11) [reply packages/shared/friends/sagas.ts:454]: Nope, we're not getting anything from it. Do you think it'd be more appropriate to hardcode the response to the renderer? I mean sth like: 

`  const updateUnseenMessages: UpdateUserUnseenMessagesPayload = {
    userId: ownId,
    total: 0
 }`

`getUnityInstance().UpdateUserUnseenMessages(updateUnseenMessages)`

File: packages/unity-interface/BrowserInterface.ts [hunk 7]
```diff
@@ -409,6 +409,14 @@ export class BrowserInterface {
     getFriendRequests(getFriendRequestsPayload)
   }
 
+  public async MarkMessagesAsSeen(userId: string) {
+    await markAsSeenPrivateChatMessages(userId)
+  }
```
Comment (menduz) [inline packages/unity-interface/BrowserInterface.ts:414]: nobody catches the failures of this async code.

please change it for:
```suggestion
  public MarkMessagesAsSeen(userId: string) {
    markAsSeenPrivateChatMessages(userId).catch(properHandler)
  }
```

**Review by jmoguilevsky — APPROVED** [review]

**Pull request merged**